File: app/src/main/java/qlsctanhoa/hcm/ditagis/com/qlsc/adapter/FeatureViewMoreInfoAdapter.java
package qlsctanhoa.hcm.ditagis.com.qlsc.adapter;

import android.content.Context;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.util.List;

import qlsctanhoa.hcm.ditagis.com.qlsc.R;

/**
 * Created by ThanLe on 04/10/2017.
 */

public class FeatureViewMoreInfoAdapter extends ArrayAdapter<FeatureViewMoreInfoAdapter.Item> {
    private Context mContext;
    private List<Item> items;

    public FeatureViewMoreInfoAdapter(Context context, List<Item> items) {
        super(context, 0, items);
        this.mContext = context;
        this.items = items;
    }

    public List<Item> getItems() {
        return items;
    }

    public void clear() {
        items.clear();
    }

    @Override
    public int getCount() {
        return items.size();
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }


//    public Item getItem(String mlt) {
//        for (Item item : this.items)
//            if (item.getTieuThu().equals(mlt))
//                return item;
//        return null;
//    }
//
//    public boolean removeItem(String mlt) {
//        for (Item item : this.items)
//            if (item.getTieuThu().equals(mlt)) {
//                this.items.remove(item);
//                return true;
//            }
//        return false;
//    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.item_viewmoreinfo, null);
        }
        Item item = items.get(position);

        LinearLayout layout = (LinearLayout) convertView.findViewById(R.id.layout_viewmoreinfo);

        if (position % 2 == 1) {
            convertView.setBackgroundColor(ContextCompat.getColor(mContext, R.color.colorBackground_1));
        } else {
            convertView.setBackgroundColor(ContextCompat.getColor(mContext, R.color.colorAccent_1));
        }
        TextView txtID = (TextView) convertView.findViewById(R.id.txt_viewmoreinfo_id);
        //todo
        txtID.setText(item.getAlias());

        TextView txtDiaChi = (TextView) convertView.findViewById(R.id.txt_viewmoreinfo_value);
        //todo
        txtDiaChi.setText(item.getValue());


        return convertView;
    }


    public static class Item {
        private String alias;
        private String value;
        private String fieldName;
        private boolean isEdit;

        public Item() {
        }

        public String getFieldName() {
            return fieldName;
        }

        public void setFieldName(String fieldName) {
            this.fieldName = fieldName;
        }

        public boolean isEdit() {
            return isEdit;
        }

        public void setEdit(boolean edit) {
            isEdit = edit;
        }

        public String getAlias() {
            return alias;
        }

        public void setAlias(String alias) {
            this.alias = alias;
        }

        public String getValue() {
            return value;
        }

        public void setValue(String value) {
            this.value = value;
        }
    }
}

File: app/src/main/java/vn/ditagis/com/tanhoa/qlsc/adapter/FeatureViewMoreInfoAdapter.kt
package vn.ditagis.com.tanhoa.qlsc.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import com.esri.arcgisruntime.data.Field

import vn.ditagis.com.tanhoa.qlsc.R

/**
 * Created by ThanLe on 04/10/2017.
 */

class FeatureViewMoreInfoAdapter(private val mContext: Context, var dItems: MutableList<Item>?) : ArrayAdapter<FeatureViewMoreInfoAdapter.Item>(mContext, 0, dItems) {

    fun getItems(): List<Item>? {
        return dItems
    }

//    fun addAll(dItems: MutableList<Item>) {
//        this.dItems = dItems
//    }

    override fun clear() {
        dItems!!.clear()
    }

    override fun getCount(): Int {
        return dItems!!.size
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    @SuppressLint("InflateParams")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        if (view == null) {
            val inflater = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(R.layout.item_viewmoreinfo, null)
        }
        val item = dItems!![position]

        val txtAlias = view!!.findViewById<TextView>(R.id.txt_viewmoreinfo_alias)
        txtAlias.text = item.alias

        val txtValue = view.findViewById<TextView>(R.id.txt_viewmoreinfo_value)
        if (item.fieldName == "ViTri" || item.fieldName == "GhiChu" || item.fieldName == "GhiChuVatTu") {
            txtValue.width = 550
        }
        txtValue.text = item.value
        if (item.isEdit) {
            view.setBackgroundColor(ContextCompat.getColor(mContext, R.color.colorAccent_1))
            view.findViewById<View>(R.id.img_viewmoreinfo_edit).visibility = View.VISIBLE
        } else {
            view.setBackgroundColor(ContextCompat.getColor(mContext, R.color.colorBackground_1))
            view.findViewById<View>(R.id.img_viewmoreinfo_edit).visibility = View.INVISIBLE

        }
        if (item.value == null)
            txtValue.visibility = View.GONE
        else
            txtValue.visibility = View.VISIBLE
        return view
    }


    class Item {
        var alias: String? = null
        var value: String? = null
        var fieldName: String? = null
        var isEdit: Boolean = false
        var fieldType: Field.Type? = null
        var isEdited: Boolean = false

        constructor()

        constructor(alias: String, value: String, fieldName: String, isEdit: Boolean, fieldType: Field.Type, isEdited: Boolean) {
            this.alias = alias
            this.value = value
            this.fieldName = fieldName
            this.isEdit = isEdit
            this.fieldType = fieldType
            this.isEdited = isEdited
        }
    }
}
